### PR TITLE
Update composer dependencies

### PR DESCRIPTION
**Pull request by rjmackay**
This pull request makes the following changes:
- Move tests out of application/ into tests/
- Update composer packages to recent versions
  - Update PHPUnit
  - Update Behat
  - Update PHPSpec
  - Update aura/di to remove custom fork
  - Switch flysystem to 1.0 not 1.1 dev
  - Update phinx
- Update phpunit tests for new version
- Update behat tests for new version
- Disable auto resolve in aura/di
- Fix DI setup to work without auto resolve
- Fix linting
- Fix bin/ushahidi to work with newer symfony/console

Future work
- Update guzzle
- Update league/uri (or replace with laravel package)
- Update phpcs

Test these changes by:
- Run tests

Refs #1372 #1340 #1654 

Ping @ushahidi/platform

---
This change is [Reviewable](https://reviewable.io/reviews/ushahidi/platform/1666)

### CODE REVIEW

File: application/classes/Ushahidi/Core.php [hunk 1]
```diff
@@ -149,7 +149,6 @@ public static function init()
 		$di->setter['Ushahidi\Console\Application']['injectCommands'][] = $di->lazyNew('Ushahidi_Console_Webhook');
 		$di->setter['Ushahidi_Console_Webhook']['setDatabase'] = $di->lazyGet('kohana.db');
 		$di->setter['Ushahidi_Console_Webhook']['setPostRepo'] = $di->lazyGet('repository.post');
-		$di->setter['Ushahidi_Console_Webhook']['setSigner'] = $di->lazyGet('tool.signer');
```
Comment (jasonmule) [inline application/classes/Ushahidi/Core.php]: Why can't this be injected here?

**Comment by jasonmule** [comment]
Updates and tests seem to work. Why do we disable auto resolve in aura/di?

**Comment by rjmackay** [comment]
> Updates and tests work ok. Why do we disable auto resolve in aura/di?

Aura/di docs recommend you turn it off because its hard to use/debug. As it happens I turned it off and found some of our DI set up was working by chance, because we had typos in dependencies etc

File: application/classes/Ushahidi/Core.php (hunk 1 above)
Comment (rjmackay) [inline application/classes/Ushahidi/Core.php]: Its unneeded, but also the signer needs an auth token param thats only set during executing of the webhook command

**Comment by rjmackay** [comment]
This is ready to land once ushahidi/platform-cloud-interface#136 is done

**Comment by rjmackay** [comment]
Ah.. So coverage drops with this because we're no longer include `vendor/` in the coverage. It shouldn't have been included in the first place so...

**Comment by willdoran** [comment]
@rjmackay @tuxpiper was saying that staging is already at 5.6 and prod should move over this week. I'd like to merge this branch as I need the changes to be able to add intercom, as such I think it should be ok for develop to go to 5.6 while master is waiting. Does that work for you? 
Should we lower the code coverage requirement since vendor caused a drop?

**Comment by rjmackay** [comment]
I think its OK to merge. Code coverage will probably fluctuate with lumen integration

**Pull request merged**